ls1intum/Artemis
# `General`: Improve exercise header

### Checklist
#### General


#### Client

### Motivation and Context


The exercise header did not show all relevant information.

### Description

The exercise header now includes more information (assessment Type, individual duedate for complaints, example solution publication date, ...).
The deadline for complaints is only shown once it's fixed (i.e. the student received the assessment from the tutor)

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming exercise, 1 Modeling exercise, 1 Quiz

1. Log in to Artemis
2. Create the exercises
3. Look at the header of each exercise 
4. Add/remove some configurations (e.g. a duedate) and check that all currently relevant information is shown
5. Also check that it looks good on small/wide screens

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots


Old:
<img width="1680" alt="Bildschirmfoto 2022-05-19 um 16 54 33" src="https://user-images.githubusercontent.com/38322605/169326620-c6ee3cef-6690-4f9a-827a-a523bb4e49b9.png">
New:
![Bildschirmfoto 2022-06-08 um 21 18 58](https://user-images.githubusercontent.com/38322605/172699082-414a681b-fb2f-409d-b660-96a3cc5c8ddb.png)

Worstcase of most possible elements in the header:
<img width="1680" alt="Bildschirmfoto 2022-06-13 um 14 58 50" src="https://user-images.githubusercontent.com/38322605/173359845-5fd20e50-a74c-41cd-8970-11d72510f512.png">
<img width="1680" alt="Bildschirmfoto 2022-06-13 um 14 59 50" src="https://user-images.githubusercontent.com/38322605/173359856-fd598842-2019-4155-9c7d-49d647444acd.png">

Worstcase on mobile:
<img width="576" alt="Bildschirmfoto 2022-06-13 um 14 59 11" src="https://user-images.githubusercontent.com/38322605/173359880-58de33cb-5045-40f4-bf8c-5f9a84007fac.png">
